docs/extend/plugins.md

```diff
@@ -1,87 +1,272 @@
 <!--[metadata]>
 +++
-title = "Extending Engine with plugins"
-description = "How to add additional functionality to Docker with plugins extensions"
-keywords = ["Examples, Usage, plugins, docker, documentation, user guide"]
+aliases = [
+"/engine/extend/"
+]
+title = "New Plugin System"
```

Comment:
We should create a new file (eg. plugins_overview.md) instead of reusing the old file. Reusing the old file preserves the git history which is confusing.